src/serializers/type_serializers/model.rs

```diff
@@ -110,7 +110,7 @@ fn has_extra(schema: &PyDict, config: Option<&PyDict>) -> PyResult<bool> {
 impl ModelSerializer {
     fn allow_value(&self, value: &PyAny, extra: &Extra) -> PyResult<bool> {
         match extra.check {
-            SerCheck::Strict => Ok(value.get_type().is(self.class.as_ref(value.py()))),
+            SerCheck::Strict => value.is_instance(self.class.as_ref(value.py())),
```

Comment:
I think in strict mode we allow subclasses during validation (we had some discussion about this and a controversial twitter poll). I am thinking the same logic should imply that we allow subclasses during serialization as well.